pyuvsim/telescope.py

```diff
@@ -37,3 +41,278 @@ def __eq__(self, other):
+                self.string_mode = False
+            else:
+                raise ValueError("Invalid beam list: " + str(beam_list))
+
+        # If any UVBeam objects are passed in, update uvb_params:
+        list_uvb_params = {}
+        if uvb_params is None:
+            uvb_params = {}
+            if not self.string_mode:
+                list_uvb_params = self._scrape_uvb_params(self._obj_beam_list, strict=False)
+        self.uvb_params.update(list_uvb_params)
+        self.uvb_params.update(uvb_params)    # Optional parameters for the UVBeam objects
+
+    def _scrape_uvb_params(self, beam_objs, strict=True):
+        """
+        Collect uvb_params from the set of beam objects.
+
+        Parameters
+        ----------
+        beam_objs : list
+            If any UVBeams are found in this list, they will be scraped
+            for relevant parameters.
+
+        strict : bool
+            If True, will raise an error if the UVBeams in beam_objs
+            have conflicting parameters.
+
+        Notes
+        -----
+        If beam_objs has strings in it, this will simply return an empty dict.
+        """
+
+        new_pars = {}
+        for bm in beam_objs:
+            if isinstance(bm, UVBeam):
+                for key in self.uvb_params.keys():
+                    val = getattr(bm, key)
+                    if key not in new_pars:
+                        new_pars[key] = []
+                    if val is not None:
+                        new_pars[key].append(val)
+        for key, vals in new_pars.items():
+            svals = set(vals)
+            if len(svals) == 1:
+                new_pars[key] = svals.pop()
+            elif len(svals) > 1:
+                try:
+                    raise ValueError('Conflicting settings for {}: {}'.format(key, str(svals)))
+                except ValueError as err:
+                    if strict:
+                        raise err
+                    else:
+                        warnings.warn(err.args[0])  # Use exception message as warning message.
+
+        new_pars = {k: v for k, v in new_pars.items() if not v == []}
+
+        return new_pars
+
+    def _set_params_on_uvbeams(self, beam_objs):
+        """
+        Set parameters on UVBeam objects using uvb_params.
+        """
+        for bm in np.atleast_1d(beam_objs):
+            if isinstance(bm, UVBeam):
+                for key, val in self.uvb_params.items():
+                    setattr(bm, key, val)
+
+    def __len__(self):
+        # Note that only one of these lists has nonzero length at a given time.
+        return len(self._obj_beam_list) + len(self._str_beam_list)
+
+    def __getitem__(self, ind):
+        if self.string_mode:
+            return self._str_beam_list[ind]
+        return self._obj_beam_list[ind]
+
+    def __setitem__(self, ind, value):
+        """
+        Insert into the beam list, converting objects to strings if in object mode,
+        or vice versa if in string mode.
```

Comment:
This docstring will error with flake8 when we turn on docstring testing in Steven's PR -- The first line should be a single line sentence (ending in a period). More details should follow in text after a blank line.